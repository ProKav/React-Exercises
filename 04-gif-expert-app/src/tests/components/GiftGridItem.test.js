import React from 'react'
import GiftGridItem from '../../components/GiftGridItem'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'

describe('Pruebas en <GiftGridItem />', () => {
	const title = 'Mi titulo'
	const url = 'https://localhost/algo.jpg'
	const wrapper = shallow(<GiftGridItem title={title} url={url} />)
	// beforeEach(() => {
	// 	wrapper = shallow(<GiftGridItem />)
	// })

	test('Debe mostrar <GiftGridItem /> correctamente', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('Debe tener un parrafo con el Title', () => {
		const p = wrapper.find('p')
		expect(p.text().trim()).toBe(title)
	})

	test('Debe tener la imagen igual al url y alt de los props', () => {
		const img = wrapper.find('img')
		expect(img.prop('alt')).toBe(title)
		expect(img.prop('src')).toBe(url)
	})

	test('Debe tener la clase animate__fadeIn', () => {
		const div = wrapper.find('div')
		const clase = div.prop('className')
		expect(clase.includes('animate__fadeIn')).toBe(true)
	})
})
