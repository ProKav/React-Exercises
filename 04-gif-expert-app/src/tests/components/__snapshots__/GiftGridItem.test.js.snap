// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGridItem /> Debe mostrar <GiftGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Mi titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Mi titulo
  </p>
</div>
`;
